presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -291,6 +300,12 @@ public boolean supportClientTransaction()
         return ImmutableSet.copyOf(splitter.split(nullToEmpty(servletRequest.getHeader(PRESTO_CLIENT_TAGS))));
     }
 
+    private Set<String> parseClientCapabilities(HttpServletRequest servletRequest)
+    {
+        Splitter splitter = Splitter.on(',').trimResults().omitEmptyStrings();
+        return ImmutableSet.copyOf(splitter.split(nullToEmpty(servletRequest.getHeader(PRESTO_CLIENT_CAPABILITIES))));
```

Comment:
Do we want to support clients putting arbitrary strings in this header? Or do we just want to fail loudly?

Reply:
"arbitrary" may actually mean "forward-compatible", newer client talking to older server